src/modules/controllers/reception.controllers.js

```diff
@@ -7,50 +7,187 @@ const tokenVerify = (token) => {
   return decoded;
 }
 
+const sort = (result, sortLable, sortDirection) => {
+  switch (sortDirection) {
+    case 'asc':
+      result.sort((a, b) =>
+        a[sortLable] < b[sortLable]
+          ? -1
+          : a[sortLable] > b[sortLable]
+            ? 1
+            : 0
+      );
+      break;
+
+    case 'desc':
+      result.sort((a, b) =>
+        a[sortLable] > b[sortLable]
+          ? -1
+          : a[sortLable] < b[sortLable]
+            ? 1
+            : 0
+      );
+      break;
+
+    default:
+      break;
+  }
+
+  return result;
+}
+
+const filter = (result, firstDate, lastDate) => {
+  if (firstDate && !lastDate) {
+    result = result.filter(element => element.date >= firstDate);
+  } else if (!firstDate && lastDate) {
+    result = result.filter(element => element.date <= lastDate);
+  } else if (firstDate && lastDate) {
+    result = result.filter(element =>
+      (element.date >= firstDate) &&
+      (element.date <= lastDate)
+    );
+  }
+
+  return result;
+}
+
 module.exports.getAllReceptions = (req, res) => {
```

Comment:
please use $or, $and for filer in request  (mongoose where condition)
and sort for sorting data